libs/feature-workspaces/src/lib/projects/projects.component.html

```diff
@@ -1,20 +1,38 @@
-<div *ngIf="workspace$|async as workspace">
-  <mat-list *ngIf="workspace.projects.length > 0; else empty">
-    <ng-container *ngFor="let p of workspace.projects">
-      <mat-list-item [ngClass]="p.projectType" [disableRipple]="true">
-        <div class="project-type" mat-line>{{ p.projectType }}</div>
-        <div class="project-name" mat-line>{{ p.name }}</div>
-        <div class="project-root" mat-line *ngIf="p.root">
-          <mat-icon>folder_open</mat-icon>
-          {{ p.root }}
+<ng-container *ngIf="workspace$|async as workspace">
+  <div>
+    <mat-list *ngIf="workspace.projects.length > 0; else empty">
+      <ng-container *ngFor="let p of workspace.projects">
+        <mat-list-item [ngClass]="p.projectType" [disableRipple]="true">
+          <div class="project-type" mat-line>{{ p.projectType }}</div>
+          <div class="project-name" mat-line>{{ p.name }}</div>
+          <div class="project-root" mat-line *ngIf="p.root">
+            <mat-icon>folder_open</mat-icon>
+            {{ p.root }}
+          </div>
+          <div fxLayoutWrap fxFlex="shrink" fxLayout="row" fxLayoutGap="12px" fxLayoutAlign="end center">
+            <button *ngFor="let a of p.actions" mat-stroked-button [routerLink]="a.link">
+              {{ a.actionDescription }}
+            </button>
+          </div>
+        </mat-list-item>
+        <mat-divider></mat-divider>
+      </ng-container>
+    </mat-list>
+  </div>
+</ng-container>
+
+<ng-template #empty>
+  <ui-dialog>
+      <img src="assets/code-icon.svg" height="180px">
+      <div class="headline">Empty Project!</div>
+      <div class="sub-headline">
+        Your project is empty. Create a new Angular application.
+      </div>
+      <button mat-raised-button color="primary" routerLink="../generate">
```

Comment:
I think we can maybe be a bit more specific here and navigate directly to the application generation route (or at least pre-filter the list of generation schematics)

Reply:
Oh I think I see what you mean. You probably meant pre-populating the filter input on the generate UI, right?